explorer/interpreter/type_checker.cpp

```diff
@@ -3089,7 +3099,8 @@ auto TypeChecker::TypeCheckExp(Nonnull<Expression*> e,
         }
         default:
           return ProgramError(e->source_loc())
-                 << "member access, unexpected " << object_type << " in " << *e;
+                 << "member access into unexpected type `" << object_type
+                 << "` in `" << *e << "`";
```

Comment:
Just a minor note: wrapping type information with backticks looks clean in console output but it isn't pretty to look at in source code, like in this example. Should I consider implementing a utility function to wrap type info in backticks, like `std::quoted`?

Reply:
I'm agnostic about this, I don't really see the backticks in source code as an issue, and a utility class would likely end up more verbose in code. I also think we could end up wanting the backticks wrapping more than one thing, as with:

```
      note_stream << sep << "`" << param->name();
      if (auto value = param->constant_value()) {
        note_stream << " = " << **value;
      }
      note_stream << "`";
```

Reply:
> I don't really see the backticks in source code as an issue

My only real concern is that they are easy to miss, but I get all of your points and I also thinks we should not introduce new functions.